test/interpreter/function-npv.spec.ts

```diff
@@ -0,0 +1,42 @@
+import {ErrorType, HyperFormula} from '../../src'
+import {ErrorMessage} from '../../src/error-message'
+import {adr, detailedError} from '../testUtils'
+
+describe('Function NPV', () => {
+  it('should return #NA! error with the wrong number of arguments', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=NPV(1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should return correct value', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=NPV(1, NA())'],
+      ['=NPV(1, B2:C2)', 1, 'abcd'],
+      ['=NPV(1, B3:C3)', true, 1],
+      ['=NPV(1, 1, "abcd")'],
+      ['=NPV(1, TRUE(), 1)'],
+      ['=NPV(-1, 0)'],
+      ['=NPV(-1,1)'],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA))
+    expect(engine.getCellValue(adr('A2'))).toEqual(0.5)
+    expect(engine.getCellValue(adr('A3'))).toEqual(0.5)
+    expect(engine.getCellValue(adr('A4'))).toEqual(detailedError(ErrorType.VALUE, ErrorMessage.NumberCoercion))
+    expect(engine.getCellValue(adr('A5'))).toEqual(0.75)
+    expect(engine.getCellValue(adr('A6'))).toEqual(0)
+    expect(engine.getCellValue(adr('A7'))).toEqual(detailedError(ErrorType.NUM, ErrorMessage.NaN))
+  })
+
+  /**
+   * Discrepancy with products #1 and #2.
+   */
+  it('cell reference', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=NPV(1,B1)', true]
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqual(0.5) //Should be 0.
```

Comment:
For all of them. They treat cell reference as 1-length range (hence no-coerce), we treat reference as value (hence we coerce).

Reply:
Could you please also add this to known limitations? For example, sth like:
"we immediately instantiate references to single cells to their values instead of treating them as 1-length ranges, which slightly changes behavior of some functions (e.g. NPV)"